### PR TITLE
WT-8316 format backward compatibility mode fixes

### PR DESCRIPTION
**Pull request by keithbostic**

### CODE REVIEW

File: test/format/config.c [hunk 1]
```diff
@@ -715,51 +716,48 @@ static void
...
     /*
      * Direct I/O may not work with backups, doing copies through the buffer cache after configuring
      * direct I/O in Linux won't work. If direct I/O is configured, turn off backups.
      */
-    if (GV(BACKUP)) {
-        if (config_explicit(NULL, "backup"))
-            testutil_die(EINVAL, "direct I/O is incompatible with backup configurations");
-        config_single(NULL, "backup=off", false);
-    }
+    DIO_CHECK("disk.backup", BACKUP);
```
Comment (jiechenbo) [inline test/format/config.c:742]: believe we don't have disk.backup, and it's called `backup`.

**Review by jiechenbo — APPROVED** [review]
LGTM! Just one requested change.

**Review by sueloverso — APPROVED** [review]
LGTM

File: test/format/config.c (hunk 1 above)
Comment (keithbostic) [reply test/format/config.c]: Thank you, @jiechenbo, fixed in 246503404.

**Pull request merged**